src/mwindow.c

```diff
@@ -242,9 +240,38 @@ static int git_mwindow_close_lru(git_mwindow_file *mwf)
 	return 0;
 }
 
+/*
+ * Close the file that contains the least recently used window. Called under
+ * lock from new_window.
```

Comment:
Does this strategy make sense? It could be that this file does have the LRU window, but it could also at the same time have the MRU window as well. So we'd be pulling out the carpet under our feet if we were to close that.

A different strategy would be to scan each file for the MRU window and then chosing the LRU window across these.

Sorry if it's what you're doing here already, I find the mwindow code extremely hard to read.

Reply:
nope, i did the naïve LRU.

and now that i take another look, the current implementation could close a file that has an in-use window! fix to that incoming, and while i'm at it, i'll do the LR-MRU algorithm you suggested.